### DISCUSSION TITLE
Ssr queries are also being fetched on client side

### DISCUSSION THREAD

**Discussion by teovillanueva** [urql-graphql/urql#2705] [Q&A]
When navigating between pages urql is sending client side GraphQL requests and the ssr is also sending the requests

![image](https://user-images.githubusercontent.com/41754896/192621772-14349e84-923c-4102-ac2a-82e60221200b.png)

I've checked and in the profile.json inside `__NEXT_DATA__` the urqlState has the fullfilled queries so there should be no call to graphql in the client side

This is my HOC
```tsx
const withGraphQLClient = withUrqlClient(
  (ssrExchange, ctx) => ({
    url:
      typeof window === 'undefined'
        ? process.env.NEXT_PUBLIC_GRAPHQL_API_ENDPOINT!
        : '/api/proxy/graphql',
    exchanges: [
      devtoolsExchange,
      dedupExchange,
      cacheExchange,
      ssrExchange,
      authExchange({
        async getAuth() {
          return ctx?.req?.headers.cookie;
        },
        addAuthToOperation({ authState, operation }) {
          if (!authState) {
            return operation;
          }

          const fetchOptions =
            typeof operation.context.fetchOptions === 'function'
              ? operation.context.fetchOptions()
              : operation.context.fetchOptions || {};

          return makeOperation(operation.kind, operation, {
            ...operation.context,
            fetchOptions: {
              ...fetchOptions,
              headers: {
                ...fetchOptions.headers,
                cookie: authState,
              },
            },
          });
        },
      }),
      fetchExchange,
    ],
  }),
  { ssr: true }
)
```

**Comment by teovillanueva** [comment]
Hello, I don't mean to be bother, but can anyone help me with this? 🙏🏻

**Comment by JoviDeCroock** [comment]
SSR: true means you are relying on the built-in ssr-prepass rendering, while `profile.json` implies you are  working with `getServerSideProps` or `getStaticProps`, [docs](https://formidable.com/open-source/urql/docs/advanced/server-side-rendering/#nextjs). Just mentioning this as it implies you are already doing double work.

It is possible that we are missing a state [here](https://github.com/FormidableLabs/urql/blob/main/packages/next-urql/src/with-urql-client.ts#L74) to restore subsequent server-side fetched pages

**Reply by teovillanueva** [reply]
Thanks for answering, should I open an issue?

BTW: im using next 12.2.5
